plugins/catalog-backend/src/service/util.ts

```diff
@@ -40,6 +45,22 @@ export async function requireRequestBody(req: Request): Promise<unknown> {
   return body;
 }
 
+export function setRequiredEntityLocationAnnotations(entity: Entity): Entity {
+  const clonedEntity = lodash.cloneDeep(entity);
+  lodash.set(
+    clonedEntity,
+    `metadata.annotations.['${LOCATION_ANNOTATION}']`,
+    'validate:entity',
+  );
+  lodash.set(
+    clonedEntity,
+    `metadata.annotations.['${ORIGIN_LOCATION_ANNOTATION}']`,
+    'validate:entity',
+  );
```

Comment:
From the [docs for `LOCATION_ANNOTATION`](https://backstage.io/docs/features/software-catalog/well-known-annotations#backstageiomanaged-by-location), I see it is added automatically by the catalog. Is it required to set this before validating the entity with the `CatalogProcessingOrchestrator`? 

Was `ORIGIN_LOCATION_ANNOTATION` also required, or is it being set since the docs indicate it is the same value as `LOCATION_ANNOTATION` in most cases?

Reply:
I'm not sure at what point those annotations added when ingesting entities but the `CatalogProcessingOrchestrator` expects them both to be set in order to be a valid entity.

Reply:
Yep they're required indeed. I'm a bit curious, did you try out a flow where these are supplied from the caller, with real values? At least as an option that seems sensible, and perhaps even a requirement to begin with. For example, if you are validating an entity definition that you intend to register from `https://github.com/backstage/backstage/blob/master/catalog-info.yaml`, you'd actually use that as the `url` location for both of these?

Thinking there is that potentially allows for more accurate validation, as some processors rely on these annotations for context, for example the `PlaceholderProcessor` that lets you use `$text` etc.

Reply:
That is an interesting idea. I've been tossing the idea of being able to pick a repository to load the `catalog-info.yaml` file from. The entity validator plugin would then validate that file and be able to commit those changes. In that case those annotations could be set but that was just ideating on what the entity validator could be in the future, not the first version.

Reply:
What I'm thinking is that you'd have to supply the intended location of the entity as part of the validation request. I'm thinking that will greatly increase compatibility with various processors that rely on the location of the entity to function.

Is it something you think you could try out/implement as part of this PR?

Reply:
Yeah I can do that, would it be more desirable to add it to part of the request body like this

```json
{
  "entity": {
    ...
  }
  "location": "url:http://github.com/backstage/backstage/blob/master/catalog-info.yaml"
}
```
or to have the location set inside the entity like this

```yaml
...
metadata:
  annotations:
    backstage.io/managed-by-location: url:http://github.com/backstage/backstage/blob/master/catalog-info.yaml
...
```

I'm inclined to do the first because of this in the docs

> This annotation is added automatically by the catalog as it fetches the data from a registered location, and is not meant to normally be written by humans.

Also, do we need to have separate values for `backstage.io/managed-by-location` and `backstage.io/managed-by-origin-location` or can we use the same value for both since most of the time they are the same value?

Reply:
@Rugvip I added a location to the request body.